File: evolution/rust-underconstruction/00-rust-fifth/evo-figures/src/project/rectangle.rs
//==============================================================================
// rectangle.rs - содержит описание прямоугольника
//==============================================================================
//------------------------------------------------------------------------------
// прямоугольник
pub struct Rectangle {
  pub x: i32,  // ширина
  pub y: i32,  // высота
}

impl Rectangle {
  //------------------------------------------------------------------------------
  // Вывод параметров прямоугольника в форматируемый поток
  pub fn out(&self) {
    println!("It is Rectangle: x = {}, y = {}", self.x, self.y);
  }
}

// Вывод параметров прямоугольника в форматируемый поток
pub fn out(r: &Rectangle) {
  println!("It is Rectangle: x = {}, y = {}", r.x, r.y);
}

File: other/time-tests/01-start-task/python/simple-string/rectangle.py
#----------------------------------------------
def Read(rect, strArray, i):
    # должно быть как минимум два непрочитанных значения в массиве
    if i >= len(strArray) - 1:
        return 0
    rect.append("rectangle")
    rect.append(int(strArray[i]))
    rect.append(int(strArray[i+1]))
    i += 2
    #print("Rectangle: x = ", rect[1], " y = ", rect[2])
    return i

def Print(rect):
    print("Rectangle: x = ", rect[1], " y = ", rect[2], ", Perimeter = ", Perimeter(rect))
    pass

def Write(rect, ostream):
    ostream.write("Rectangle: x = {}  y = {}, Perimeter = {}".format(rect[1], rect[2], Perimeter(rect)))
    pass

def Perimeter(rect):
    return 2.0 * (rect[1] + rect[2])
    pass
